### PR TITLE
Geom: nil guard a few rect methods

### LINKED ISSUE #7656: Koreader crashes on Kobo when opening menu after opening book when wifi is on
* KOReader version: 2021.04
* Device: Kobo Glo HD (kobo_alyssum)

#### Steps to reproduce

1. Ensure wifi is turned on
2. Open file browser
3. Open any book
4. Immediately after opening the book, try to open main menu (by swiping down from the top edge of the screen

BUG: Koreader crashes

IMG_20210509_092943

Issue does not appear to happen if either Wifi is off or if you turn a book page before opening menu.

##### `crash.log` (if applicable)

[crash.log](https://github.com/koreader/koreader/files/6447094/crash.log)

### PR DESCRIPTION
**Pull request by NiLuJe**
We've managed to trip a few of those on dimen fields post-init but
pre-paintTo in a few weird corner-cases, a point at which dimen is often
nil.

ConfigDialog: Deal with that very thing in update()

Fix #7656

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7664)

### CODE REVIEW

File: frontend/apps/filemanager/filemanager.lua [hunk 1]
```diff
@@ -539,6 +539,15 @@ function FileManager:init()
 
     self:initGesListener()
     self:handleEvent(Event:new("SetDimensions", self.dimen))
+
+    -- NOTE: ReaderUI has a _getRunningInstance method for this, because it used to store the instance reference in a private module variable.
+    if FileManager.instance == nil then
+        logger.dbg("Spinning up new FileManager instance", tostring(self))
+    else
+        -- Should never happen, given what we did above...
+        logger.warn("FileManager instance mismatch! Opened", tostring(self), "while we still have an existing instance:", tostring(FileManager.instance))
```
Comment (Frenzie) [inline frontend/apps/filemanager/filemanager.lua:548]: You could use a dbg.err or something to make sure we actually notice it.

**Review by poire-z — APPROVED** [review]

File: frontend/apps/filemanager/filemanager.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/apps/filemanager/filemanager.lua]: @NiLuJe I actually meant the straight up crash dbg.dassert variety like:

https://github.com/koreader/koreader/blob/ea3fa5c2c7ed7330e8e1e6ef838b780e7998ea51/plugins/batterystat.koplugin/main.lua#L244

Comment (NiLuJe) [reply frontend/apps/filemanager/filemanager.lua]: Huh. Missed that one when going through Dbg (which, err, could use some documentation ^^).

Comment (Frenzie) [reply frontend/apps/filemanager/filemanager.lua]: I was being rather vague I suppose. I mainly meant anything along the lines of dbg.guard/dbg.dassert/dbg.whatever that you feel is appropriate but hard crashes the program so I would actually notice if or when it happens. ;-)

At the same time this isn't a very important issue, hopefully.

Comment (Frenzie) [reply frontend/apps/filemanager/filemanager.lua]: For future reference: https://github.com/koreader/koreader/pull/7677

Comment (NiLuJe) [reply frontend/apps/filemanager/filemanager.lua]: Yeah, that one branch should *really* be dead code ;).

**Pull request merged**